library/n2_protocol.hpp

```diff
@@ -37,12 +45,34 @@ struct message {
 	virtual ~message() = default;
 };
 
+class call {
```

Comment:
2 -- Ok. I ordered methods according to call lifecycle, but it only one viewpoint.
1 -- I doubt should they be `const`, because some requests (e.g. `write_request`) are high-weighted, and should be passed as mutable values (to move some data from request, instead of copying it). Maybe it's better to use `unique_ptr` instead of raw pointers here. I'll try to use.

Reply:
What about try to use unique_ptr? Got it?
p.s.: `get_request`still be not const

Reply:
I can use `unique_ptr` for future, but this time I have to `release` it immedialelly, because pointer then goes through C code.

Reply:
Please, make `get_request` method `const` too.